extensions/amp-sidebar/0.1/amp-sidebar.js

```diff
@@ -132,10 +133,23 @@ export class AmpSidebar extends AMP.BaseElement {
     this.registerAction('open', this.open_.bind(this));
     this.registerAction('close', this.close_.bind(this));
 
+    // TODO(mkhatib, #6589): Consider exposing onLocalNavigation from
+    // document-click service to simplifiy this.
     this.element.addEventListener('click', e => {
       const target = closestByTag(dev().assertElement(e.target), 'A');
       if (target && target.href) {
-        this.close_();
+        const tgtLoc = parseUrl(target.href);
+        const curLoc = parseUrl(this.getAmpDoc().win.location.href);
+        const tgtHref = `${tgtLoc.origin}${tgtLoc.pathname}${tgtLoc.search}`;
```

Comment:
Done. Should i keep the check for `tgt.hash` still or just close if the navigation is not to a new page? I think it's not entirely needed but a bit safer and we can relax it moving forward.